app/controllers/katello/api/v2/products_controller.rb

```diff
@@ -50,7 +50,10 @@ def index_relation
       query = query.enabled if params[:enabled]
       query = query.where(:id => @activation_key.products) if @activation_key
       query = query.where(:id => @system.products) if @system
-      query = query.where(:id => Pool.find_by_id!(params[:subscription_id]).products) if params[:subscription_id]
+      if params[:subscription_id]
+        pool = Pool.with_identifiers(params[:subscription_id])
```

Comment:
This would technically be returning `pools` vs a single pool? Might be worth having a singular version of with_identifier that re-uses the logic from with_identifiers